classes/migration/install/MetricsMigration.inc.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * @file classes/migration/install/MetricsMigration.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class MetricsMigration
+ * @brief Describe database table structures.
+ */
+
+namespace APP\migration\install;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema as Schema;
+
+class MetricsMigration extends \PKP\migration\Migration
+{
+    /**
+     * Run the migrations.
+     * This migration file is used during upgrades. If this schema changes, the upgrade scripts should be reviewed manually before a merging.
+     */
+    public function up(): void
+    {
+        Schema::create('metrics_context', function (Blueprint $table) {
+            $table->string('load_id', 255);
+            $table->bigInteger('context_id');
+            $table->date('date');
+            $table->integer('metric');
+            $table->foreign('context_id')->references('journal_id')->on('journals');
+            $table->index(['load_id'], 'metrics_context_load_id');
+            $table->index(['context_id'], 'metrics_context_context_id');
+        });
+        Schema::create('metrics_submission', function (Blueprint $table) {
+            $table->string('load_id', 255);
+            $table->bigInteger('context_id');
+            $table->bigInteger('submission_id');
+            $table->bigInteger('representation_id')->nullable();
+            $table->bigInteger('submission_file_id')->unsigned()->nullable();
+            $table->bigInteger('file_type')->nullable();
+            $table->bigInteger('assoc_type');
```

Comment:
This is because it is so much easier to construct those 'group by assoc type' in the query builder, to get the numbers for different assoc types at once. In metrics_issue I do not use it, but then I need to convert the 'group by assoc type' into a complicated and not so understandable SQL (s. StatsIssueQueryBuilder::getSelectColumns()). And assoc types are even more complicated in metrics_submission (and for OMP with chapters even more).
Back then I asked Alec, he thought it would be OK this way, but... Maybe once he had a look at the whole code, we can see...

Reply:
We'll probably always have a few cases where this is necessary, but they are very few and I'd like to get rid of `assoc_type`/`assoc_id` when possible. I'm OK for it to stay in this case.

(The equivalent in Laravel land is "morphs"; eventually we may adopt that.)

Reply:
:+1:  I would then like to leave it here...